subsys/bluetooth/mesh/proxy_msg.c

```diff
@@ -230,6 +241,46 @@ int bt_mesh_proxy_relay_send(struct bt_conn *conn, struct bt_mesh_adv *adv)
 	return err;
 }
 
+int bt_mesh_proxy_relay_send(struct bt_conn *conn, struct bt_mesh_adv *adv)
+{
+#if defined(CONFIG_BT_MESH_PROXY_MSG_ALWAYS_SEND)
+	if (k_current_get() == k_work_queue_thread_get(&k_sys_work_q)) {
+		struct bt_mesh_proxy_role *role = &roles[bt_conn_index(conn)];
+
+		k_fifo_put(&role->pending, bt_mesh_adv_ref(adv));
+		bt_mesh_wq_submit(&role->work);
+
+		return 0;
+	}
+#endif /* CONFIG_BT_MESH_PROXY_MSG_ALWAYS_SEND */
+
+	return proxy_relay_send(conn, adv);
```

Comment:
I have also considered this, but I think **relay** messages are not in order, because from the perspective of radio propagation, there is a possibility that the actual message sent later may be received first because the actual message sent earlier is interfered with.

Reply:
> I wonder if we try another approach and try to back-off when the host API returns an error which per my understanding (or please correct me if I missing something) can now happen if the host ran out of cmd buffers. If this worked then we wouldn't need an extra thread and could just try to reschedule the advertiser and send the same adv again (same for proxy).

It has been modified based on this, so the issue of out-of-order becomes even less important.